src/ray/core_worker/core_worker.cc

```diff
@@ -1783,31 +1794,51 @@ Status CoreWorker::GetLocationFromOwner(
     return Status::OK();
   }
 
+  absl::flat_hash_map<rpc::Address, std::vector<ObjectID>> objects_by_owner;
+  for (const auto &object_id : object_ids) {
+    rpc::Address owner_address;
+    RAY_RETURN_NOT_OK(GetOwnerAddress(object_id, &owner_address));
+    objects_by_owner[owner_address].push_back(object_id);
+  }
+
   auto mutex = std::make_shared<absl::Mutex>();
-  auto num_remaining = std::make_shared<size_t>(object_ids.size());
+  auto num_remaining = std::make_shared<size_t>(objects_by_owner.size());
   auto ready_promise = std::make_shared<std::promise<void>>();
   auto location_by_id =
       std::make_shared<absl::flat_hash_map<ObjectID, std::shared_ptr<ObjectLocation>>>();
 
-  for (const auto &object_id : object_ids) {
-    rpc::Address owner_address;
-    RAY_RETURN_NOT_OK(GetOwnerAddress(object_id, &owner_address));
+  for (const auto &pair : objects_by_owner) {
+    const auto &owner_address = pair.first;
+    const auto &owner_object_ids = pair.second;
     auto client = core_worker_client_pool_->GetOrConnect(owner_address);
     rpc::GetObjectLocationsOwnerRequest request;
```

Comment:
Currently Ray grpc max message size is 512 MB. In theory, if we have lots of objects for a single owner, the request size can exceeds that limit. Although it should be very rare, can we set a batch size limit just to be safe: e.g. a single batch can have at most 100 objects.